### PR TITLE
v1.10 backports 2022-06-13

### PR DESCRIPTION
**Pull request by nebril**
* ~~#18348 -- metrics: Add extra clustermesh metrics (@sayboras)~~
 * #19522 -- redirectpolicy: Fix panic in service matcher LPR processing (@aditighag)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 19522; do contrib/backporting/set-labels.py $pr done 1.10; done
```
or with
```
$ make add-label BRANCH=v1.10 ISSUES=19522
```

### CODE REVIEW

**Comment by nebril** [comment]
/test-backport-1.10

**Comment by sayboras** [comment]
Seems like there is compilation issue in metrics.go.

**Comment by nebril** [comment]
/test-backport-1.10

File: pkg/logging/logfields/logfields.go [hunk 1]
```diff
@@ -300,7 +300,7 @@ const (
 	// Devices is the devices name
 	Devices = "devices"
 
-	//DirectRoutingDevice is the name of the direct routing device
+	// DirectRoutingDevice is the name of the direct routing device
```
Comment (aditighag) [inline pkg/logging/logfields/logfields.go:303]: Same as 1.9 backport comment - https://github.com/cilium/cilium/pull/20180/commits/49295384be0f75e0752a9aae0a3b180d202c06a1#r896169706.

Comment (pchaigno) [reply pkg/logging/logfields/logfields.go:303]: I don't think it matters enough to repush to be honest.

**Comment by pchaigno** [comment]
Runtime failed with https://github.com/cilium/cilium/issues/19950; k8s-1.19-kernel-4.9 failed with https://github.com/cilium/cilium/issues/19055 (fixed on master and v1.11 but not on v1.9 and v1.10). Aditi reviewed her changes. Merging.

**Pull request merged**